vpn/src/main/java/com/duckduckgo/mobile/android/vpn/network/RealVpnDetector.kt

```diff
@@ -32,28 +28,12 @@ import javax.inject.Inject
 @ContributesBinding(AppScope::class)
 class RealVpnDetector @Inject constructor(
     private val context: Context,
-    private val appBuildConfig: AppBuildConfig
 ) : VpnDetector {
 
     override fun isVpnDetected(): Boolean {
-        return if (appBuildConfig.sdkInt <= Build.VERSION_CODES.LOLLIPOP_MR1) {
-            detectVpnLollipop()
-        } else {
-            detectVpn()
-        }
-    }
-
-    @RequiresApi(Build.VERSION_CODES.M)
-    private fun detectVpn(): Boolean {
-        val connectivityManager = context.applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-        val activeNetwork = connectivityManager.activeNetwork
-        return connectivityManager.getNetworkCapabilities(activeNetwork)?.hasTransport(NetworkCapabilities.TRANSPORT_VPN) ?: false
-    }
-
-    @TargetApi(Build.VERSION_CODES.LOLLIPOP_MR1)
-    private fun detectVpnLollipop(): Boolean {
         val connectivityManager = context.applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-        return connectivityManager.activeNetworkInfo?.type == ConnectivityManager.TYPE_VPN
+        return connectivityManager.allNetworks
```

Comment:
This is not the same as what we had before. This loops through all the networks tracked by the framework, as opposed to the active one, which is what we want.